File: backend/CoreCuestionariosOIJ/src/CuestionariosEntidades/Models/SubCategory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CuestionariosEntidades.Models
{
    public class SubCategory
    {
        public int? Id { get; set; }
        public String? Name { get; set; } = String.Empty;
        public int? IdCategory { get; set; }
        public bool? IsDeleted { get; set; }


        // EF Relation
        public Category? Category;
        // EF Relation
        public IEnumerable<Question>? Questions { get; set; }

    }
}

File: frontend/src/app/models/Subcategory.ts
export class SubCategory{

    name?: String
    id?: number;
    idCategory?: number;
    editable?: boolean;
    
    constructor({name, id, idCategory}:{name?: String, id?: number, idCategory?: number}) {
        this.name=name;
        this.id=id;
        this.idCategory=idCategory;
        this.editable=false;
    }

}